core/src/data/clientGeoJsonSource.cpp

```diff
@@ -0,0 +1,186 @@
+#include "clientGeoJsonSource.h"
+#include "mapbox/geojsonvt/geojsonvt_convert.hpp"
+#include "mapbox/geojsonvt/geojsonvt_types.hpp"
+#include "platform.h"
+#include "util/geom.h"
+
+using namespace mapbox::util;
+
+namespace Tangram {
+
+const double extent = 4096;
+const uint8_t maxZoom = 18;
+const uint8_t indexMaxZoom = 18;
+const uint32_t indexMaxPoints = 100000;
+double tolerance = 0.01;
+
+Point ClientGeoJsonSource::transformPoint(geojsonvt::TilePoint pt) const {
+    return { 2 * pt.x / extent - 1, 1 - 2 * pt.y / extent, 0 };
+}
+
+ClientGeoJsonSource::ClientGeoJsonSource(const std::string& _name, const std::string& _url)
+    : DataSource(_name, _url) {
+
+    if (!_url.empty()) {
+        // Load from file
+        const auto& string = stringFromResource(_url.c_str());
+        addData(string);
+    }
+}
+
+void ClientGeoJsonSource::addData(const std::string& _data) {
+
+    m_features = geojsonvt::GeoJSONVT::convertFeatures(_data);
+    m_store = std::make_unique<GeoJSONVT>(m_features, maxZoom, indexMaxZoom, indexMaxPoints, tolerance);
+
+}
+
+bool ClientGeoJsonSource::loadTileData(std::shared_ptr<TileTask>&& _task, TileTaskCb _cb) {
+
+    _cb(std::move(_task));
+
+    return true;
+}
+
+bool ClientGeoJsonSource::getTileData(std::shared_ptr<TileTask>& _task) {
+
+    return true;
+}
+
+void ClientGeoJsonSource::clearData() {
+
+    m_features.clear();
+    m_store.reset();
+    DataSource::clearData();
```

Comment:
This is for just clearing the data (not at destruction time) - for instance if you want to plot a new route or set of search results.